rest/data/admin.go

```diff
@@ -0,0 +1,104 @@
+package data
+
+import (
+	"github.com/evergreen-ci/evergreen/model/admin"
+	"github.com/evergreen-ci/evergreen/model/event"
+)
+
+type DBAdminConnector struct{}
+
+// GetAdminSettings retrieves the admin settings document from the DB
+func (ac *DBAdminConnector) GetAdminSettings() (*admin.AdminSettings, error) {
+	return admin.GetSettings()
+}
+
+// SetAdminSettings sets the admin settings document in the DB and event logs it
+func (ac *DBAdminConnector) SetAdminSettings(settings admin.AdminSettings) error {
+	return UpdateSettings(settings)
+}
+
+// SetAdminBanner sets the admin banner in the DB and event logs it
+func (ac *DBAdminConnector) SetAdminBanner(text string) error {
+	return UpdateBanner(text)
+}
+
+// SetServiceFlags sets the service flags in the DB and event logs it
+func (ac *DBAdminConnector) SetServiceFlags(flags admin.ServiceFlags) error {
+	return UpdateServiceFlags(flags)
+}
+
+type MockAdminConnector struct {
+	MockSettings admin.AdminSettings
+}
+
+// GetAdminSettings retrieves the admin settings document from the mock connector
+func (ac *MockAdminConnector) GetAdminSettings() (*admin.AdminSettings, error) {
+	return &ac.MockSettings, nil
+}
+
+// SetAdminSettings sets the admin settings document in the mock connector
+func (ac *MockAdminConnector) SetAdminSettings(settings admin.AdminSettings) error {
+	ac.MockSettings = settings
+	return nil
+}
+
+// SetAdminBanner sets the admin banner in the mock connector
+func (ac *MockAdminConnector) SetAdminBanner(text string) error {
+	ac.MockSettings.Banner = text
+	return nil
+}
+
+// SetServiceFlags sets the service flags in the mock connector
+func (ac *MockAdminConnector) SetServiceFlags(flags admin.ServiceFlags) error {
+	ac.MockSettings.ServiceFlags = flags
+	return nil
+}
+
+// UpdateSettings updates the entire admin settings document and logs events
+func UpdateSettings(settings admin.AdminSettings) error {
```

Comment:
I wanted them on the admin settings object, but since they log events it would create an import loop. I moved these into the data connector for Brian's comment